### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by Nyholm**

### CODE REVIEW

File: docs/clients/lambda.md [hunk 1]
```diff
@@ -5,21 +5,19 @@ category: clients
 
 # Lambda Client
 
-```bash
-export AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE
-export AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
-```
-See [docs/authentication.md](https://github.com/async-aws/aws/blob/master/docs/authentication.md) to see how to authenticate against AWS server.
+See section about [autnetication](/docs/authentication/index.md) to learn how to
```
Comment (jderusse) [inline docs/clients/lambda.md]: authentication

Comment (Nyholm) [reply docs/clients/lambda.md]: Thank you

File: tests/Unit/DocumentationTest.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            self::assertEquals($excerpt, $startOfSection, sprintf('There is no mention of "%s"\'s default value.', $name));
+        }
+    }
+
+    public function testDocumentationLinks()
+    {
+        $invalidLinks = [
+            '(./docs/',
+            '(/docs/',
+            '(docs/',
```
Comment (Nyholm) [inline tests/Unit/DocumentationTest.php]: I made these error plenty.

File: tests/Unit/DocumentationTest.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+use Symfony\Component\Finder\SplFileInfo;
+
+/**
+ * Verify documentation bugs
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class DocumentationTest extends TestCase
+{
+    public function testConfigurationDocs()
```
Comment (Nyholm) [inline tests/Unit/DocumentationTest.php]: This makes sure we always document the configuration reference.

File: docs/configuration.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+**Default:** 'https://%service%.%region%.amazonaws.com'
+
+### roleArn
+
+???
+
+### webIdentityTokenFile
+
+???
```
Comment (jderusse) [inline docs/configuration.md]:
```
Path to the file that contains the OAuth 2.0 access token when using a web identity provider
```

File: docs/configuration.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+### roleSessionName
+
+???
```
Comment (jderusse) [inline docs/configuration.md]:
```
An identifier for the assumed role session

**Default:** 'async-aws-' followed by random chars
```

**Comment by Nyholm** [comment]
Thank you. I wrote some comments, but yours are much better.

File: docs/features/async.md [hunk 6]
```diff
@@ -4,9 +4,126 @@ category: features
...
+the remote server again.
+
+The function also has a `?float $timeout = null` argument. If the timeout is set to
+`2.0`, we will wait for 2 seconds for a response. If a response is received, the function
+will return `true` or thrown an exception. If the timeout is reached, it will return
+`false`.
+
+Use `$result->resolve(0)` for a non-blocking call.
+
+### Batch requests
```
Comment (Nyholm) [inline docs/features/async.md]: @gmorel I used your example here. Output and everything. 

Thank you

**Pull request merged**